package.json

```diff
@@ -50,32 +58,127 @@
         "path": "./syntaxes/markdown.tmLanguage.json"
       }
     ],
-    "themes": [
+    "commands": [
       {
-        "label": "D2 Light",
-        "uiTheme": "vs",
-        "path": "./themes/light-color-theme.json"
-      },
+        "command": "d2-viewer.ShowPreviewWindow",
+        "title": "Show d2 Preview Window",
+        "category": "d2"
+      }
+    ],
+    "keybindings": [
       {
-        "label": "D2 Dark",
-        "uiTheme": "vs-dark",
-        "path": "./themes/dark-color-theme.json"
+        "command": "d2-viewer.ShowPreviewWindow",
+        "key": "ctrl+shift+d",
+        "mac": "shift+cmd+d"
       }
-    ]
+    ],
+    "configuration": {
+      "type": "object",
+      "title": "d2 Viewer Configuration",
+      "properties": {
+        "d2-viewer.autoUpdate": {
+          "type": "boolean",
+          "default": true,
+          "description": "Automatically update the preview."
+        },
+        "d2-viewer.updateTimer": {
+          "type": "number",
+          "default": 1500,
+          "description": "Specifies the amount of time to wait between keystrokes before automatically updating the preview. (1000 = 1000ms = 1s)"
+        },
+        "d2-viewer.updateOnSave": {
+          "type": "boolean",
+          "default": false,
+          "description": "Automatically update on saving the d2 file.  Turn off to avoid excess updates when Auto Save is enabled."
+        }
+      }
+    },
+    "menus": {
+      "commandPalette": [
+        {
+          "command": "d2-viewer.ShowPreviewWindow",
+          "when": "editorLangId == d2",
+          "group": "d2"
+        }
+      ],
+      "editor/context": [
+        {
+          "command": "d2-viewer.ShowPreviewWindow",
+          "when": "resourceLangId == d2",
+          "group": "d2"
+        }
+      ],
+      "editor/title": [
+        {
+          "command": "d2-viewer.ShowPreviewWindow",
+          "when": "resourceLangId == d2",
+          "group": "d2"
+        }
+      ],
+      "editor/title/context": [
+        {
+          "command": "d2-viewer.ShowPreviewWindow",
+          "when": "resourceLangId == d2",
+          "group": "d2"
+        }
+      ],
+      "explorer/context": [
+        {
+          "command": "d2-viewer.ShowPreviewWindow",
+          "when": "resourceLangId == d2",
+          "group": "d2"
+        }
+      ]
+    }
   },
+  "scripts": {
+    "vscode:prepublish": "yarn run package",
+    "compile": "webpack",
+    "watch": "webpack --watch",
+    "package": "webpack --mode production --devtool hidden-source-map",
+    "compile-tests": "tsc -p . --outDir out",
```

Comment:
do we need this for the non-test ts files?

Reply:
I'll strip out all the test stuff since it doesn't do anything, the other ares needed for development work.

Reply:
Removed all the unneeded tasks in PR #16

Reply:
can we add to this PR?

Reply:
Done